pkg/database/database.go

```diff
@@ -50,8 +51,201 @@ func updateFromEnv(dsn string) (string, error) {
 	return result, nil
 }
 
+// extractPlaceholders replaces ${...} placeholders with unique sentinels that are URL-safe.
+// This allows us to parse the URL structure before expanding environment variables.
+// Returns: the string with sentinels, a mapping of sentinel->varName, and any error.
+func extractPlaceholders(s string) (string, map[string]string, error) {
+	mapping := make(map[string]string)
+	counter := 0
+
+	result := DSNREnvRegex.ReplaceAllStringFunc(s, func(match string) string {
+		sentinel := fmt.Sprintf("__PH_%d__", counter)
+		varName := match[2 : len(match)-1] // Extract VAR from ${VAR}
+		mapping[sentinel] = varName
```

Comment:
Why not look up the value here and throw it in the map? Then you don't have to go from sentinel value to variable name to actual value.

Reply:
revamping is... fine?  but is a different game.  still need to support env expansion in the DSN regardless of fully paramterizing the config?